### PR TITLE
binutils: Fix cross, again

### PR DESCRIPTION
**Pull request by Ericson2314**
###### Motivation for this change

It's been broken since I made it multi-arch, but this fixes it. Making this a separate PR to lessen the diff, and in case anything breaks later.

###### Things done

Tested exhaustively in #26805.

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/build-support/bintools-wrapper/default.nix [hunk 1]
```diff
@@ -164,6 +164,20 @@ stdenv.mkDerivation {
       set +u
     '';
 
+  emulation = let
```
Comment (Ericson2314) [inline pkgs/build-support/bintools-wrapper/default.nix]: I think I will move this to the lib, with some of the other misc per-tool stuff / provide a way for `{build,host,target}Platform` (e.g. via `crossConfig`) to override it. That, however, is a non-hash-breaking change so no reason to rush it now.

File: pkgs/development/tools/misc/binutils/default.nix [hunk 2]
```diff
@@ -53,6 +53,14 @@ stdenv.mkDerivation rec {
     # elf32-littlearm-vxworks in favor of the first.
     # https://github.com/NixOS/nixpkgs/pull/30484#issuecomment-345472766
     ./disambiguate-arm-targets.patch
+
```
Comment (Ericson2314) [inline pkgs/development/tools/misc/binutils/default.nix]: I or @bgamari should email upstream cause the default behavior is insane.

Comment (orivej) [reply pkgs/development/tools/misc/binutils/default.nix]: I have reported this at https://sourceware.org/bugzilla/show_bug.cgi?id=22458, with no response so far. (Said so in https://github.com/NixOS/nixpkgs/pull/30484#issuecomment-345491222.)

Comment (orivej) [reply pkgs/development/tools/misc/binutils/default.nix]: Ah, you meant `always-search-rpath.patch` below.

**Pull request merged**